### PR TITLE
TS-1257: [ssl_cipher_table] replace TCL HashTable by unordered_map

### PR DESCRIPTION
**Pull request by persiaAziz**
Replace the underlying TCL hash table implementation of ssl_cipher_name_table by STL unordered_map. This PR partially addresses TS-1257. There are many places where TCL hash table is being used. I plan to replace these one by one.

### CODE REVIEW

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -141,8 +142,9 @@ static int ssl_vc_index = -1;
 static ink_mutex *mutex_buf      = nullptr;
 static bool open_ssl_initialized = false;
 
-RecRawStatBlock *ssl_rsb                   = nullptr;
-static InkHashTable *ssl_cipher_name_table = nullptr;
+RecRawStatBlock *ssl_rsb = nullptr;
+typedef std::unordered_map<const char *, intptr_t> cipherTable;
```
Comment (jpeach) [inline iocore/net/SSLUtils.cc]: This hashes on the pointer value, not on the string value. I'm not sure that OpenSSL guarantees to give the same pointer values when you fish them out of the SSL context.

**Comment by SolidWallOfCode** [comment]
I think the goal is to remove all TCL dependencies, not necessarily to improve performance.

File: iocore/net/SSLUtils.cc (hunk 1 above)
Comment (zwoop) [inline iocore/net/SSLUtils.cc]: That's what I thought too (we provide custom hashing functions for this in a few places), but then I tested this on a modern compiler, and it seemed to do it right. I assume C++11 had added support for hashing the char*, but now that you say this, it's quite possible my tests were wrong.

**Comment by persiaAziz** [comment]
will make a new PR

**Pull request closed without merging**